source/common/http/http2/codec_impl.cc

```diff
@@ -849,6 +861,8 @@ ConnectionImpl::ConnectionImpl(Network::Connection& connection, CodecStats& stat
                                const uint32_t max_headers_kb, const uint32_t max_headers_count)
     : use_new_codec_wrapper_(
           Runtime::runtimeFeatureEnabled("envoy.reloadable_features.http2_new_codec_wrapper")),
+      use_oghttp2_library_(
+          Runtime::runtimeFeatureEnabled("envoy.reloadable_features.http2_use_oghttp2")),
```

Comment:
I suggest adding an assert to validate that this runtime feature is enabled while `envoy.reloadable_features.http2_new_codec_wrapper` is disabled.
Could prevent cases where users don't understand why they enable this runtime flag and nothing happens.

Reply:
Done, thank you.

Reply:
shouldn't this only be set if both guards are true?

Reply:
I added an assertion that fails if `envoy.reloadable_features.http2_use_oghttp2` is enabled while `envoy.reloadable_features.http2_new_codec_wrapper` is disabled.